lib/collection/src/operations/mod.rs

```diff
@@ -208,39 +208,54 @@ impl Validate for CollectionUpdateOperations {
     }
 }
 
-fn point_to_shard(point_id: ExtendedPointId, ring: &HashRing<ShardId>) -> ShardId {
-    *ring
-        .get(&point_id)
-        .expect("Hash ring is guaranteed to be non-empty")
+/// Get the shards for a point ID
+///
+/// Normally returns a single shard ID. Might return multiple if resharding is currently in
+/// progress.
+///
+/// # Panics
+///
+/// Panics if the hash ring is empty and there is no shard for the given point ID.
+fn point_to_shards(point_id: &ExtendedPointId, ring: &ShardHashRing) -> ShardIds {
+    let shard_ids = ring.get(point_id);
+    assert!(
+        !shard_ids.is_empty(),
+        "Hash ring is guaranteed to be non-empty",
+    );
+    shard_ids
 }
 
 /// Split iterator of items that have point ids by shard
-fn split_iter_by_shard<I, F, O>(
+fn split_iter_by_shard<I, F, O: Clone>(
     iter: I,
     id_extractor: F,
-    ring: &HashRing<ShardId>,
+    ring: &ShardHashRing,
 ) -> OperationToShard<Vec<O>>
 where
     I: IntoIterator<Item = O>,
     F: Fn(&O) -> ExtendedPointId,
 {
     let mut op_vec_by_shard: HashMap<ShardId, Vec<O>> = HashMap::new();
     for operation in iter {
-        let shard_id = point_to_shard(id_extractor(&operation), ring);
-        op_vec_by_shard.entry(shard_id).or_default().push(operation);
+        for shard_id in point_to_shards(&id_extractor(&operation), ring) {
```

Comment:
Basically, I propose we do this (in all places we use `point_to_shards`):

```suggestion
        for shard_id in point_to_shards(&id_extractor(&operation), ring).take(1) {
```

And then even if we start resharding (and initialize `ShardHashRing::Resharding`) it will not change split-by-shard behavior *yet*, it will still only split-by-shard to the old hashring.

(Maybe slight change to `ShardHashRing::get` required. I'm not sure if `dedup` preserves `old, new` order.)

Reply:
And also to `select_shards`, so that read operations are not redirected to new shard.

Reply:
It is not entirely clear to me what you suggest.

Are you suggesting to just route to the original shard for now, and enable multiple shards at a later point in time?

If so it may be easier to change `get()` to just return a single shard ID all the time. That requires a change in just one place, versus in a few places we potentially miss.

Reply:
Implemented in 52b1a22f4f691bb5b2c4d9fb675b8cdcd390eeca